pytorch_lightning/callbacks/pruning.py

```diff
@@ -0,0 +1,336 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Pruning
+^^^^^^^
+
+"""
+from copy import deepcopy
+from functools import partial
+from typing import Callable, List, Optional, Tuple, Union
+
+from torch import nn
+from torch.nn.modules.container import ModuleDict, ModuleList
+
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.utilities import _PYTORCH_PRUNE_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+if _PYTORCH_PRUNE_AVAILABLE:
+    import torch.nn.utils.prune as pytorch_prune
+
+
+_PYTORCH_PRUNING_FUNCTIONS = {
+    "ln_structured": pytorch_prune.ln_structured,
+    "l1_unstructured": pytorch_prune.l1_unstructured,
+    "random_structured": pytorch_prune.random_structured,
+    "random_unstructured": pytorch_prune.random_unstructured,
+}
+
+_PYTORCH_PRUNING_METHOD = {
+    "ln_structured": pytorch_prune.LnStructured,
+    "l1_unstructured": pytorch_prune.L1Unstructured,
+    "random_structured": pytorch_prune.RandomStructured,
+    "random_unstructured": pytorch_prune.RandomUnstructured,
+}
+
+
+def _check_parameters_to_prune(p):
+    """
+    Check the provide element is a pair with:
+        * nn.Module
+        * str
+
+    Example::
+
+        parameters_to_prune = [
+            (model.mlp_1, "weight"),
+            (model.mlp_2, "weight")
+        ]
+    """
+    return len(p) == 2 and isinstance(p[0], nn.Module) and isinstance(p[1], str)
+
+
+def check_parameters_to_prune(
```

Comment:
could you add a docstring to this function what it more precisely does?
also should it be private?

Reply:
Done !